src/Nethermind/Nethermind.Optimism/OptimismReceiptMessageDecoder.cs

```diff
@@ -12,10 +12,8 @@ namespace Nethermind.Optimism;
 public class OptimismReceiptTrieDecoder() : OptimismReceiptMessageDecoder(true) { }
 
 [Rlp.Decoder]
-public class OptimismReceiptMessageDecoder(bool isEncodedForTrie = false) : IRlpStreamDecoder<OptimismTxReceipt>, IRlpStreamDecoder<TxReceipt>
+public class OptimismReceiptMessageDecoder(bool isEncodedForTrie = false) : IRlpStreamDecoder<TxReceipt>
```

Comment:
Ok, so this is probably the main change. Instead of going with decode/encode op specific receipt and the regular one, we use the common denominator here and handle `TxReceipt` that might be (dynamic check `is`) of the optimistic type.

Reply:
As you correctly pointed out, this is the main change that avoids triggering an exception. We had a similar case in the past (https://github.com/NethermindEth/nethermind/pull/8157) where we always processed items as if the were Deposit only.